src/stableSort.js

```diff
@@ -0,0 +1,31 @@
+var _curry2 = require('./internal/_curry2');
+var _slice = require('./internal/_slice');
+
+/**
+ * Returns a copy of the list, sorted according to the comparator function, which should accept two values at a
+ * time and return a negative number if the first value is smaller, a positive number if it's larger, and zero
+ * if they are equal, values of the same key retain their original order. Please note that this is a **copy** of the list.
+ *  It does not modify the original.
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig (a,a -> Number) -> [a] -> [a]
+ * @param {Function} comparator A sorting function :: a -> b -> Int
+ * @param {Array} list The list to sort
+ * @return {Array} a new array with its elements sorted by the comparator function.
+ * @example
+ *
+ *      var diff = function(a, b) { return a - b; };
+ *      R.sort(diff, [4,2,7,5]); //=> [2, 4, 5, 7]
+ */
+module.exports = _curry2(function stableSort(comparator, list) {
+  var copyList = _slice(list);
+
+  return copyList.sort(function(a, b) {
+    if (comparator(a, b) === 0) {
+      return -(copyList.indexOf(a) - copyList.indexOf(b));
```

Comment:
An alternative to scanning the list twice with `indexOf` here would be to zip up the input list with the index of each field and then use the index to perform the secondary comparison.

e.g. something like:
```js
var stableSort = R.curry(function (comparator, list) {
  var withIndex = R.zip(list, R.range(0, list.length));
  var sortedWithIndex = withIndex.sort(function(a, b) {
    var order = comparator(a[0], b[0]);
    return order === 0 ? a[1] - b[1] : order;
  });
  return R.map(R.head, sortedWithIndex);
});
```

I'd be interested to see how these two compare for different sized lists.